src/sets.jl

```diff
@@ -80,6 +80,22 @@ function dual_set_type end
 
 dual_set_type(S::Type{<:AbstractSet}) = error("Dual type of $S is not implemented.")
 
+"""
+    distance_to_set(v, s)
+
+Compute the distance of a value to a set.
+For some vector-valued sets, can return a vector of distances.
```

Comment:
A vector of distances isn't a distance, mathematically speaking. What is meant here precisely?

Reply:
I guess we want `distance_to_set(point, set::AbstractSet; norm::Int = 2)`. Or even just say that we're using the L2 norm.

Reply:
the intuition I had for it is giving an element-wise distance when it makes sense, giving which element to "improve" to reduce the distance to the set.
We can also always normalize as @odow mentions

Reply:
If we add the distance type parameter we can easily allow the user to switch from the norm2 thing to a norm infinity which is pretty reasonable